tools/sancov/sancov.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import "fmt"
+import "flag"
+import "os"
+import "encoding/binary"
+
+const (
+	magic32SecondHalf uint32 = 0xFFFFFF32
+	magic64SecondHalf uint32 = 0xFFFFFF64
+	magicFirstHalf    uint32 = 0xC0BFFFFF
+)
+
+func usage() {
+	progName := os.Args[0]
+	fmt.Printf("Usage:\n")
+	fmt.Printf(" %s print FILE [FILE...]\n", progName)
+	os.Exit(-1)
+}
+
+func readMagicAndReturnBitness(f *os.File) uint8 {
+	header := [...]uint32{0, 0}
+	if err := binary.Read(f, binary.LittleEndian, &header); err != nil {
+		panic(err)
```

Comment:
I understand that you port this from python, but it is more civilized to print a readable error message rather than internal stack trace.
In such utilities it is usually very handy to have a failf function that prints a formatted error message to stderr and terminates process:

    failf("failed to read input file: %v", err)

Here and below.